### PR TITLE
[DOCS] Clarify that dense vectors can be created with ES

### PR DESCRIPTION
**Pull request by abdonpijpelink**
The kNN docs currently state about dense vectors that: "You create these vectors outside of Elasticsearch". However, you can also create them in ES using an NLP model. This PR clarifies that, with a link to [an example in the Machine Learning documentation](https://www.elastic.co/guide/en/machine-learning/current/ml-nlp-text-emb-vector-search-example.html).

/cc @addytripathi

### CODE REVIEW

File: docs/reference/search/search-your-data/knn-search.asciidoc [hunk 1]
```diff
@@ -20,9 +20,11 @@ Common use cases for kNN include:
 === Prerequisites
 
 * To run a kNN search, you must be able to convert your data into meaningful
-vector values. You create these vectors outside of {es} and add them to
-documents as <<dense-vector,`dense_vector`>> field values. Queries are
-represented as vectors with the same dimension.
+vector values. You can
+{ml-docs}/ml-nlp-text-emb-vector-search-example.html[create these vectors using
+a natural language processing (NLP) model in {es}], or generate them outside
+{es}, and add them to documents as <<dense-vector,`dense_vector`>> field values.
```
Comment (jrodewig) [inline docs/reference/search/search-your-data/knn-search.asciidoc:26]: Nit: This is a pretty complex sentence. I'd consider breaking it after "generate them outside {es}"

**Review by jrodewig — APPROVED** [review]
LGTM aside from a non-blocking nit. Thanks!

File: docs/reference/search/search-your-data/knn-search.asciidoc (hunk 1 above)
Comment (abdonpijpelink) [reply docs/reference/search/search-your-data/knn-search.asciidoc]: Done. Thank James!

**Review by jrodewig — APPROVED** [review]
Changes LGTM. Thanks!

**Pull request merged**